pkg/networkservice/acl/acl_parser.go

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package acl
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/edwarnicke/govpp/binapi/acl_types"
+	"github.com/edwarnicke/govpp/binapi/ip_types"
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+)
+
+const (
+	action        = "action"     // DENY, PERMIT, REFLECT
+	dstPrefix     = "dst_prefix" // IPv4 or IPv6 CIDR
+	srcPrefix     = "src_prefix" // IPv4 or IPv6 CIDR
+	protocol      = "proto"
+	icmpTypeFirst = "icmp_type_first" // 8-bit unsigned integer
+	icmpTypeLast  = "icmp_type_last"  // 8-bit unsigned integer
+	icmpCodeFirst = "icmp_code_first" // 8-bit unsigned integer
+	icmpCodeLast  = "icmp_code_last"  // 8-bit unsigned integer
+	srcPortLast   = "src_port_last"   // 16-bit unsigned integer
+	srcPortFirst  = "src_port_first"  // 16-bit unsigned integer
+	dstPortLast   = "dst_port_last"   // 16-bit unsigned integer
+	dstPortFirst  = "dst_port_first"  // 16-bit unsigned integer
+	tcpFlagsMask  = "tcp_flags_mask"  // 8-bit unsigned integer
+	tcpFlagsValue = "tcp_flags_value" // 8-bit unsigned integer
+)
+
+func parseACLRulesMap(ctx context.Context, rules map[string]string) (aclRules []acl_types.ACLRule) {
+	logger := log.FromContext(ctx).WithField("acl", "parser")
+	for _, r := range rules {
+		rule := &acl_types.ACLRule{}
+
+		parsed := parseKVStringToMap(r, ",", "=")
+
+		if err := addAction(parsed, rule); err != nil {
+			logger.Errorf("Error parsing action: %v", err.Error())
+			continue
+		}
+
+		if err := addProtocol(parsed, rule); err != nil {
+			logger.WithField("acl", "parser").Errorf("Error parsing protocol: %v", err.Error())
```

Comment:
`logger` already have added `acl:parser` field, please don't add it again.

Reply:
done